sql/parse/parse.go

```diff
@@ -671,8 +671,56 @@ func convertCreateTable(ctx *sql.Context, c *sqlparser.DDL) (sql.Node, error) {
 		}
 	}
 
+	var idxDefs []*plan.IndexDefinition
+	for _, idxDef := range c.TableSpec.Indexes {
+		if idxDef.Info.Primary {
+			continue
+		}
+
+		//TODO: add vitess support for FULLTEXT
+		constraint := sql.IndexConstraint_None
+		if idxDef.Info.Unique {
+			constraint = sql.IndexConstraint_Unique
+		} else if idxDef.Info.Spatial {
+			constraint = sql.IndexConstraint_Spatial
+		}
+
+		columns := make([]sql.IndexColumn, len(idxDef.Columns))
+		for i, col := range idxDef.Columns {
+			if col.Length != nil {
+				if col.Length.Type == sqlparser.IntVal {
+					length, err := strconv.ParseInt(string(col.Length.Val), 10, 64)
+					if err != nil {
+						return nil, err
+					}
+					if length < 1 {
+						return nil, ErrInvalidIndexPrefix.New(length)
+					}
+				}
+			}
+			columns[i] = sql.IndexColumn{
+				Name:   col.Column.String(),
+				Length: 0,
```

Comment:
we want to pass `length` here right?

Reply:
Length is used for index prefixes (https://dev.mysql.com/doc/refman/8.0/en/column-indexes.html#column-indexes-prefix). We fail on invalid input, but for now we don't have a full/proper implementation of prefixes yet, so the length just gets dropped.